flink-filesystems/flink-azure-fs-hadoop/src/main/resources/META-INF/services/org.apache.flink.core.fs.FileSystemFactory

```diff
@@ -15,3 +15,5 @@
 
 org.apache.flink.fs.azurefs.AzureFSFactory
 org.apache.flink.fs.azurefs.SecureAzureFSFactory
+org.apache.flink.fs.azurefs.ABFSAzureFSFactory
```

Comment:
Later on, I would love for us to rename these factories to something that clarifies the filesystems they actually support.

For now, it is ok. It would just be a minor improvement for clarity @AHeise should we make that a FLIP?

- org.apache.flink.fs.azurefs.AzureFSFactory supports wasb scheme (Non-Secure Azure Blob Storage)
- org.apache.flink.fs.azurefs.SecureAzureFSFactory supports wasbs scheme (Secure Azure Blob Storage)
- org.apache.flink.fs.azurefs.ABFSAzureFSFactory supports abfs scheme (Non-Secure Azure Data Lake Store Gen 2)
- org.apache.flink.fs.azurefs.SecureABFSAzureFSFactory supports abfss scheme (Secure Azure Data Lake Store Gen 2)

We could rename them as follows:

- org.apache.flink.fs.azurefs.AzureFSFactory -> org.apache.flink.fs.azurefs.AzureBlobStorageFSFactory
- org.apache.flink.fs.azurefs.SecureAzureFSFactory -> org.apache.flink.fs.azurefs.SecureAzureBlobStorageFSFactory 
- org.apache.flink.fs.azurefs.ABFSAzureFSFactory -> org.apache.flink.fs.azurefs.AzureDataLakeStoreGen2FSFactory
- org.apache.flink.fs.azurefs.SecureABFSAzureFSFactory -> org.apache.flink.fs.azurefs.SecureAzureDataLakeStoreGen2FSFactory

What do you think @srinipunuru @AHeise ?

Reply:
It's not user-facing, so a FLIP is not needed at all. In fact, I'm assuming that it would be best to do it immediately here. Rename the old factories in 1 additional commit in this PR and rename the new factories in the main commit of this PR.

Reply:
Good feedback. I updated the file names in the latest change.